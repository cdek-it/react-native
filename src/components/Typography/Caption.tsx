import React from 'react'
import { Text, type TextProps } from 'react-native'

import { makeStyles } from '../../utils/makeStyles'

export interface CaptionProps extends TextProps {
  color?: 'default' | 'secondary' | 'primary'
  disabled?: boolean
}

export const Caption = ({ color = 'default', disabled, style, ...other }: CaptionProps) => {
  const styles = useStyles()

  return (
    <Text style={[styles.text, styles[color], disabled && styles.disabled, style]} {...other} />
  )
}

const useStyles = makeStyles(({ theme }) => ({
  text: {
    fontSize: 12.25,
    includeFontPadding: false,
    verticalAlign: 'middle',
  },
  default: {
    color: theme.General.textColor,
  },
  primary: {
    color: theme.General.primaryColor,
  },
  secondary: {
    color: theme.General.textSecondaryColor,
  },
  disabled: {
    opacity: 0.6,
  },
}))
