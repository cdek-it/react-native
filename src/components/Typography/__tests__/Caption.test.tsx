import { render } from '@testing-library/react-native'

import { Caption, type CaptionProps } from '../Caption'

describe('Caption component tests', () => {
  const snapshotCases = generatePropsCombinations<CaptionProps>({
    color: ['default', 'secondary', 'primary'],
    disabled: [true, false],
  })

  test.each(snapshotCases)('color = $color, disabled = $disabled', (props) => {
    const renderedCaption = render(<Caption {...props}>Text</Caption>)

    expect(renderedCaption.toJSON()).toMatchSnapshot()
  })

  test('default props', () => {
    const renderedCaption = render(<Caption>Text</Caption>)

    expect(renderedCaption.toJSON()).toMatchSnapshot()
  })
})
