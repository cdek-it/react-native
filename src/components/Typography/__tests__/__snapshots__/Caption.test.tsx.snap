// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Caption component tests color = default, disabled = false 1`] = `
<Text
  style={
    [
      {
        "fontSize": 12.25,
        "includeFontPadding": false,
        "verticalAlign": "middle",
      },
      {
        "color": "rgba(0, 0, 0, 0.8000)",
      },
      false,
      undefined,
    ]
  }
>
  Text
</Text>
`;

exports[`Caption component tests color = default, disabled = true 1`] = `
<Text
  style={
    [
      {
        "fontSize": 12.25,
        "includeFontPadding": false,
        "verticalAlign": "middle",
      },
      {
        "color": "rgba(0, 0, 0, 0.8000)",
      },
      {
        "opacity": 0.6,
      },
      undefined,
    ]
  }
>
  Text
</Text>
`;

exports[`Caption component tests color = primary, disabled = false 1`] = `
<Text
  style={
    [
      {
        "fontSize": 12.25,
        "includeFontPadding": false,
        "verticalAlign": "middle",
      },
      {
        "color": "#188700",
      },
      false,
      undefined,
    ]
  }
>
  Text
</Text>
`;

exports[`Caption component tests color = primary, disabled = true 1`] = `
<Text
  style={
    [
      {
        "fontSize": 12.25,
        "includeFontPadding": false,
        "verticalAlign": "middle",
      },
      {
        "color": "#188700",
      },
      {
        "opacity": 0.6,
      },
      undefined,
    ]
  }
>
  Text
</Text>
`;

exports[`Caption component tests color = secondary, disabled = false 1`] = `
<Text
  style={
    [
      {
        "fontSize": 12.25,
        "includeFontPadding": false,
        "verticalAlign": "middle",
      },
      {
        "color": "rgba(0, 0, 0, 0.6000)",
      },
      false,
      undefined,
    ]
  }
>
  Text
</Text>
`;

exports[`Caption component tests color = secondary, disabled = true 1`] = `
<Text
  style={
    [
      {
        "fontSize": 12.25,
        "includeFontPadding": false,
        "verticalAlign": "middle",
      },
      {
        "color": "rgba(0, 0, 0, 0.6000)",
      },
      {
        "opacity": 0.6,
      },
      undefined,
    ]
  }
>
  Text
</Text>
`;

exports[`Caption component tests default props 1`] = `
<Text
  style={
    [
      {
        "fontSize": 12.25,
        "includeFontPadding": false,
        "verticalAlign": "middle",
      },
      {
        "color": "rgba(0, 0, 0, 0.8000)",
      },
      undefined,
      undefined,
    ]
  }
>
  Text
</Text>
`;
